cms/static/js/certificates/views/add_course_mode.js

```diff
@@ -0,0 +1,70 @@
+define([
+    'underscore',
+    'gettext',
+    'js/views/baseview',
+    'common/js/components/views/feedback_notification',
+    'text!templates/course-modes.underscore',
+    'edx-ui-toolkit/js/utils/html-utils'
+],
+function(_, gettext, BaseView, NotificationView, CourseModes, HtmlUtils) {
+    'use strict';
+
+    var AddCourseMode = BaseView.extend({
+        el: $('.wrapper-certificates'),
+        events: {
+            'click .add-course-mode': 'addCourseMode'
+        },
+
+        initialize: function(options) {
+            this.enableCourseModeCreation = options.enableCourseModeCreation;
+            this.courseModeCreationUrl = options.courseModeCreationUrl;
+            this.courseId = options.courseId;
+        },
+
+        render: function() {
+            HtmlUtils.setHtml(this.$el, HtmlUtils.template(CourseModes)({
+                enableCourseModeCreation: this.enableCourseModeCreation,
+                courseModeCreationUrl: this.courseModeCreationUrl,
+                courseId: this.courseId
+            }));
+            return this;
+        },
+
+        addCourseMode: function() {
+            var notification = new NotificationView.Mini({
+                title: gettext('Enabling honor course mode')
+            });
+            var username = $('.account-username')[0].innerText;
+            $.ajax({
+                url: this.courseModeCreationUrl + '?username=' + username,
```

Comment:
note to author: how can we update the course mode API so we don't necessarily have to send the username